### PR TITLE
Add Result.tryPeek

### LINKED ISSUE #13: Add Result.tryPeek
as an exception handling alternative for `Result.peek`.

### PR DESCRIPTION
**Pull request by grevend-bosch**
Blocked by: #8

Signed-off-by: David Greven <fixed-term.David.Greven@de.bosch.com>

### CODE REVIEW

File: common-types/src/main/java/io/carbynestack/common/result/Success.java [hunk 1]
```diff
@@ -105,6 +106,29 @@ public boolean isSuccess() {
...
+     * @since 0.1.0
+     */
+    @Override
+    public Result<S, F> tryPeek(AnyThrowingConsumer<? super S> consumer, F reason) {
+        requireNonNull(consumer);
+        try {
+            consumer.accept(this.value());
+            return this;
+        } catch (Throwable throwable) {
+            return new Failure<>(reason);
```
Comment (sbckr) [inline common-types/src/main/java/io/carbynestack/common/result/Success.java:128]: All information about the exception that occurred is lost and there is no way to recover the message or cause.
While the `reason` of the error might be sufficient to provide error details for a simple `consumer` logic, logging or further handling must be done in complex consumers themselves.

How about something like making `reason` capable of holding the Throwable.

File: common-types/src/main/java17/io/carbynestack/common/result/Success.java [hunk 2]
```diff
@@ -75,6 +77,29 @@ public boolean isSuccess() {
...
+     * @since 0.1.0
+     */
+    @Override
+    public Result<S, F> tryPeek(AnyThrowingConsumer<? super S> consumer, F reason) {
+        requireNonNull(consumer);
+        try {
+            consumer.accept(this.value());
+            return this;
+        } catch (Throwable throwable) {
+            return new Failure<>(reason);
```
Comment (sbckr) [inline common-types/src/main/java17/io/carbynestack/common/result/Success.java:99]: See comment on [common-types/src/main/java/io/carbynestack/common/result/Success.java](common-types/src/main/java/io/carbynestack/common/result/Success.java)

**Review by sbckr — COMMENTED** [review]
Please see comment on [common-types/src/main/java/io/carbynestack/common/result/Success.java](common-types/src/main/java/io/carbynestack/common/result/Success.java)

File: common-types/src/main/java/io/carbynestack/common/result/Success.java (hunk 1 above)
Comment (grevend-bosch) [reply common-types/src/main/java/io/carbynestack/common/result/Success.java:128]: From a technical point of view, it is already possible to save a `Throwable` in reason, but this special implementation of peek currently deals with the simplest case (and the only one we have used so far). As soon as the described (more complex) case becomes relevant, we can take a similar path as with `Result.of`.

**Review by sbckr — APPROVED** [review]
LGTM

**Pull request merged**